### DISCUSSION TITLE
`GenericAlias` is not compatible with `type`?

### DISCUSSION THREAD

**Discussion by shcheklein** [python/typing#1843] [Q&A]
Trying to understand if this is a bug or not:

`a: type = list[int]`

works fine, but:

`a: type = list[type(1)]`

gives:

```python
test.py:2:11: error: Type expected within [...]  [misc]
    a: type = list[type(1)]
              ^
test.py:2:11: error: Incompatible types in assignment (expression has type "GenericAlias", variable has type "type")  [assignment]
    a: type = list[type(1)]
              ^~~~~~~~~~~~~
Found 2 errors in 1 file (checked 1 source file)
```

**Comment by sobolevn** [comment]
Yes, it is expected. MyPy does not evaluate `type()` call in this case.

**Reply by shcheklein** [reply]
Why does it have to evaluate it though? `type` returns `Type` I assume? (we don't need to know the exact type `int` in this case, no?).

Also, why do we have the second error?  `list[int]` is also has type of `GenericAlias` AFAIU, and it assigns it fine to `a: type`.

**Reply by erictraut** [reply]
I would consider this a bug in mypy. This is a value expression, not a type expression, so mypy should not be applying rules that apply to type expressions. By comparison, [pyright](https://pyright-play.net/?pyrightVersion=1.1.374&code=IYLgBALgngDgpmAvGANgSwM4QNrXgCgEYBKAXQCgg) does not generate an error here.

In mypy's defense, the typing spec is silent on this topic. Draft [PEP 747](https://peps.python.org/pep-0747/) may help provide additional clarity about this matter.

**Comment by Aalian123** [comment]
Using type in type hints confusing the type checker. It expects a different type or generic type not type evaluations inside type annotations.
